bft/bft_types/src/lib.rs

```diff
@@ -89,6 +90,137 @@ impl fmt::Display for PositionedInstruction {
         write!(f, "{}:{} {}", self.line, self.character, self.instruction)
     }
 }
+/// Instructions that have been processed into a form useful to an interpreter
+pub enum DecoratedInstruction {
```

Comment:
This type is quite complex, I'm not sure why you need it to be this much.  eg. do you need the `closer` and `opener` as well as the indices?